common/repositories/inventory_snapshots_repository.h

```diff
@@ -68,6 +95,227 @@ class InventorySnapshotsRepository: public BaseInventorySnapshotsRepository {
 
 		return count;
 	}
+
+	static void ClearCharacterInvSnapshots(Database &db, uint32 character_id, bool from_now)
+	{
+		uint32 del_time = time(nullptr);
+		if (!from_now) {
+			del_time -= RuleI(Character, InvSnapshotHistoryD) * 86400;
+		}
+
+		DeleteWhere(db, fmt::format("`character_id` = '{}' AND `time_index` <= '{}'", character_id, del_time));
+	}
+
+	static void ListCharacterInvSnapshots(Database &db, uint32 character_id, std::list<std::pair<uint32, int>> &is_list)
+	{
+		const std::string &query = fmt::format(
+			"SELECT `time_index`, COUNT(*) FROM `inventory_snapshots` WHERE "
+			"`character_id` = '{}' GROUP BY `time_index` ORDER BY `time_index` DESC",
+			character_id
+		);
+		auto results = db.QueryDatabase(query);
+
+		if (!results.Success())
+			return;
+
+		for (auto row: results) {
+			is_list.emplace_back(std::pair<uint32, int>(Strings::ToUnsignedInt(row[0]), Strings::ToInt(row[1])));
+		}
+	}
+
+	static bool ValidateCharacterInvSnapshotTimestamp(Database &db, uint32 character_id, uint32 timestamp)
+	{
+		if (!character_id || !timestamp) {
+			return false;
+		}
+
+		const std::string &query = fmt::format(
+			"SELECT * FROM `inventory_snapshots` WHERE `character_id` = '{}' "
+			"AND `time_index` = '{}' LIMIT 1",
+			character_id,
+			timestamp
+		);
+		auto results = db.QueryDatabase(query);
+
+		if (!results.Success() || results.RowCount() == 0) {
+			return false;
+		}
+
+		return true;
+	}
+
+	static void ParseCharacterInvSnapshot(
+		Database &db,
+		uint32 character_id,
+		uint32 timestamp,
+		std::list<std::pair<int16, uint32>> &parse_list)
+	{
+		const std::string &query = fmt::format(
+			"SELECT `slot_id`, `item_id` FROM `inventory_snapshots` "
+			"WHERE `character_id` = '{}' AND `time_index` = '{}' ORDER BY `slot_id`",
```

Comment:
Remove quotes from numbers in query please

Reply:
Updated